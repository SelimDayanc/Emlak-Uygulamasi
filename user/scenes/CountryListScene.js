/*
 @flow
 */
import React, {Component, PropTypes} from 'react';
import {
  ListView,
  StyleSheet,
  Text,
  TouchableHighlight,
  View,
} from 'react-native';
import colors from '../../common/colors';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import {CountryPropType} from '../../property/common/proptypes';

export default class CountryListScene extends Component {
  static propTypes = {
    countries: PropTypes.array.isRequired,
    onCountrySelect: PropTypes.func.isRequired,
    country: CountryPropType.isRequired,
  };

  constructor(props) {
    super(props);
    this.ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 != r2});
  }

  renderRow = item => {
    let {onCountrySelect, country} = this.props;
    return (
      <View style={styles.row} key={item}>
        <TouchableHighlight
          onPress={() => onCountrySelect(item)}
          underlayColor="transparent">
          <View style={styles.rowContent}>
            <Text
              style={[
                styles.title,
                country.fullName === item && {
                  color: colors.accent,
                  fontWeight: '500',
                },
              ]}>
              {' '}{item}{' '}
            </Text>
          </View>
        </TouchableHighlight>
      </View>
    );
  };

  renderSeparator = (sectionID: number, rowID: number) => {
    return <View style={styles.separator} key={`${sectionID}-${rowID}`} />;
  };

  render() {
    const {countries} = this.props;
    let dataSource = this.ds.cloneWithRows(countries);

    return (
      <ListView
        style={styles.container}
        dataSource={dataSource}
        renderRow={this.renderRow}
        renderSeparator={this.renderSeparator}
        enableEmptySections={true}
        ref="listView"
        showsVerticalScrollIndicator={false}
        showsHorizontalScrollIndicator={false}
        automaticallyAdjustContentInsets={false}
        removeClippedSubviews={false}
      />
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
  },
  row: {
    flex: 1,
    padding: 20,
  },
  rowContent: {
    flex: 1,
    alignItems: 'center',
    flexDirection: 'row',
  },
  title: {
    flex: 1,
    color: colors.darkGrey,
    fontWeight: '200',
  },
  separator: {
    flex: 1,
    height: 0.5,
    backgroundColor: colors.lightGrey,
  },
});
