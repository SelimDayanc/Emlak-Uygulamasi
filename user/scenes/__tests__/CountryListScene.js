import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CountryListScene from '../../scenes/CountryListScene';

test('renders CountryListScene', () => {
  const tree = renderer
    .create(
      <CountryListScene
        country={{
          fullName: 'Kuwait',
          abbr: 'KW',
          currency: 'KD',
          coords: {
            latitude: 29.3759,
            longitude: 47.9774,
          },
        }}
        onCountrySelect={() => {}}
        countries={['Kuwait', 'Bahrain', 'Saudi Arabia']}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
