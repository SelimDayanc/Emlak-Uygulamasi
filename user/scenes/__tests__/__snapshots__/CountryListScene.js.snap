// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CountryListScene 1`] = `
<RCTScrollView
  alwaysBounceHorizontal={undefined}
  alwaysBounceVertical={true}
  automaticallyAdjustContentInsets={false}
  dataSource={
    ListViewDataSource {
      "items": 3,
    }
  }
  enableEmptySections={true}
  onContentSizeChange={null}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={undefined}
  onResponderTerminationRequest={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onScrollShouldSetResponder={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  removeClippedSubviews={false}
  renderRow={[Function]}
  renderScrollComponent={[Function]}
  renderSeparator={[Function]}
  sendMomentumEvents={false}
  showsHorizontalScrollIndicator={false}
  showsVerticalScrollIndicator={false}
  style={
    Array [
      Object {
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "overflow": "scroll",
      },
      Object {
        "backgroundColor": "white",
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollContentView
    collapsable={false}
    removeClippedSubviews={false}
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "padding": 20,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#484848",
                  "flex": 1,
                  "fontWeight": "200",
                },
                Object {
                  "color": "#ff1144",
                  "fontWeight": "500",
                },
              ]
            }
          >
             
            Kuwait
             
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "padding": 20,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#484848",
                  "flex": 1,
                  "fontWeight": "200",
                },
                false,
              ]
            }
          >
             
            Bahrain
             
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "padding": 20,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#484848",
                  "flex": 1,
                  "fontWeight": "200",
                },
                false,
              ]
            }
          >
             
            Saudi Arabia
             
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollContentView>
</RCTScrollView>
`;
